File: src/ProjectR.Scripting/content/scripts/spells/AcidicSea.lua
function GetTargetType()
    return Enemies
end

function GetName()
return "Acidic Sea"
end

function GetDescription()
return "A sea of acid surrounds the enemy.\nLowers enemies AD."
end

function IsSupportSpell()
    return false
end

function GetMPCost()
    return .6
end

function GetDelay()
  return .3
end

function GetSpellType()
  return Magical
end

function GetMasteries()
  return { WAT }
end

function SpellEffect()
  local damage = (3.0625 * aMD * (aWAT/100) - .875 * dMR) * (100/dWAT)
  d:BuffStat(AD, -0.18)
  d:TakeDamage(damage);
end

File: src/ProjectR.Scripting/content/scripts/spells/AcidicSea.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting
{
    public class AcidicSea : SpellScriptBase
    {
        public override string Name { get { return "Acidic Sea"; } }
        public override string Description { get { return "A sea of acid surrounds the enemy.\nLowers enemies AD."; } }

        public override TargetType TargetType { get { return TargetType.Enemies; } }
        public override IList<EleMastery> Masteries { get { return new[] { EleMastery.WAT }; } }
        public override SpellType SpellType { get { return SpellType.Magical; } }
        public override bool IsSupportSpell { get { return false; } }

        public override double MPCost { get { return 60; } }
        public override double Delay { get { return .3; } }

        public override void SpellEffect(ICharacter caster, IList<ICharacter> targets)
        {
            foreach (var target in targets)
            {
                var damage = (3.0625 * MD(caster) * (WAT(caster) / 100) - .875 * MR(target)) * (100 / WAT(target));
                BuffStat(target, Stat.AD, -.18);
                DealDamage(target, damage);
            }
        }
    }
}